solutions/devsprint-gabriel-goto-1/app/src/main/java/com/devpass/spaceapp/presentation/NextLaunchesViewModel.kt

```diff
@@ -13,8 +13,9 @@ class NextLaunchesViewModel : ViewModel() {
 
     private val api by lazy { InitializerApi.getSpaceAPI() }
 
-    private val _apiResult = MutableLiveData<NetworkResult<LaunchPageResponse>>()
-    val apiResult: MutableLiveData<NetworkResult<LaunchPageResponse>> = _apiResult
+    private val _apiResult = MutableLiveData<NetworkResult<LaunchPageResponse>>(NetworkResult.Loading())
```

Comment:
Criativo rs, mas caso ocorra um erro de api, o retry nao iria trilar outro loading, e é interessante o loading sempre ser mostrado quando uma api for chamado

seria interesante remos ele então antes da API

```kotlin
            _apiResult.value = NetworkResult.Loading()

val state: NetworkResult<LaunchPageResponse> = isSuccess {
                api.fetchNextLaunches(QueryParams(OptionsRequest(20)))
            }
            _apiResult.value = state
```